src/main/java/duke/main/Duke.java

```diff
@@ -0,0 +1,151 @@
+
+        //Step 2. Formatting the window to look as expected
+        stage.setTitle("Duke");
+        stage.setResizable(false);
+        stage.setMinHeight(600.0);
+        stage.setMinWidth(400.0);
+
+        mainLayout.setPrefSize(400.0, 600.0);
+
+        scrollPane.setPrefSize(385, 535);
+        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
+        scrollPane.setVbarPolicy(ScrollPane.ScrollBarPolicy.ALWAYS);
+
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        // You will need to import `javafx.scene.layout.Region` for this.
+        dialogContainer.setPrefHeight(Region.USE_COMPUTED_SIZE);
+
+        userInput.setPrefWidth(325.0);
+
+        sendButton.setPrefWidth(55.0);
+
+        AnchorPane.setTopAnchor(scrollPane, 1.0);
+
+        AnchorPane.setBottomAnchor(sendButton, 1.0);
+        AnchorPane.setRightAnchor(sendButton, 1.0);
+
+        AnchorPane.setLeftAnchor(userInput, 1.0);
+        AnchorPane.setBottomAnchor(userInput, 1.0);
+
+        //Step 3. Add functionality to handle user input.
+        sendButton.setOnMouseClicked((event) -> handleUserInput());
+
+        userInput.setOnAction((event) -> handleUserInput());
+
+        //Scroll down to the end every time dialogContainer's height changes.
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+    }
+
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(getResponse(userInput.getText()));
+        dialogContainer.getChildren().addAll(
+            DialogBox.getUserDialog(userText.toString(), new ImageView(user).getImage()),
+            DialogBox.getDukeDialog(dukeText.toString(), new ImageView(duke).getImage())
+        );
+        userInput.clear();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        String output = Parser.parseCommand(input, taskList);
+
+        if (output.equals("BYE")) {
+            return exitDuke(taskList);
+        } else {
+            return output;
+        }
+    }
+
+    //Custom exitDuke Method to handle data saving and timed termination
+    private String exitDuke(TaskList taskList) {
+        //Save new data to file before exiting
+        try {
+            storage.save(taskList);
+        } catch (CannotSaveFileException e) {
+            return e.toString();
+        }
+
+        //Timed exit to terminate the program while allowing you to see the Bye Message
+        Executors.newSingleThreadScheduledExecutor()
+            .schedule(() -> System.exit(0), 2, TimeUnit.SECONDS);
```

Comment:
Indentation for wrapped lines should be 8 spaces, the dot should be below 'r' if i didn't count wrongly.

Reply:
I have IntelliJ ensuring that my checkstyle.xml is being followed, and this is indentation is automatic by IntelliJ. I've also run the validation one more time, and there are currently no checkstyle violations - but thank you for checking nonetheless!